packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogRumPlugin.kt

```diff
@@ -275,6 +286,222 @@ class DatadogRumPlugin(
+                    event.view.referrer = it["referrer"] as? String
+                    event.view.url = it["url"] as String
+                }
+
+                event
+            }
+        }
+    }
+
+    @Suppress("ComplexMethod")
+    internal fun mapErrorEvent(event: ErrorEvent): ErrorEvent? {
+        var jsonEvent = event.toJson().asMap()
+        jsonEvent = extractExtraUserInfo(jsonEvent)
+
+        return callEventMapper("mapErrorEvent", event, jsonEvent) { encodedResult, event ->
+            if (encodedResult == null) {
+                null
+            } else {
+                (encodedResult?.get("error") as? Map<String, Any?>)?.let { encodedError ->
+                    val encodedCauses = encodedError["causes"] as? List<Map<String, Any?>>
+                    if (encodedCauses != null) {
+                        event.error.causes?.let { causes ->
+                            if (causes.count() == encodedCauses.count()) {
+                                causes.forEachIndexed { i, cause ->
+                                    cause.message = encodedCauses[i]["message"] as? String ?: ""
+                                    cause.stack = encodedCauses[i]["stack"] as? String
+                                }
+                            }
+                        }
+                    } else {
+                        event.error.causes = null
+                    }
+
+                    val encodedResource = encodedError["resource"] as? Map<String, Any?>
+                    if (encodedResource != null) {
+                        event.error.resource?.url = encodedResource["url"] as? String ?: ""
+                    }
+
+                    event.error.stack = encodedError["stack"] as? String
+                }
+
+                (encodedResult?.get("view") as? Map<String, Any?>)?.let {
+                    event.view.name = it["name"] as? String
+                    event.view.referrer = it["referrer"] as? String
+                    event.view.url = it["url"] as String
+                }
+
+                event
+            }
+        }
+    }
+
+    internal fun mapLongTaskEvent(event: LongTaskEvent): LongTaskEvent? {
+        var jsonEvent = event.toJson().asMap()
+        jsonEvent = extractExtraUserInfo(jsonEvent)
+
+        return callEventMapper("mapLongTaskEvent", event, jsonEvent) { encodedResult, event ->
+            if (encodedResult == null) {
+                null
+            } else {
+
+                (encodedResult?.get("view") as? Map<String, Any?>)?.let {
+                    event.view.name = it["name"] as? String
+                    event.view.referrer = it["referrer"] as? String
+                    event.view.url = it["url"] as String
+                }
+
+                event
+            }
+        }
+    }
+
+    private fun extractExtraUserInfo(encodedEvent: Map<String, Any?>): Map<String, Any?> {
+        val reservedKeys = listOf("email", "id", "name")
```

Comment:
minor: doesn't really matter for such small collection, but for lookup it is better to use `setOf`